### PR TITLE
[feat] Unit tests

### PR DESCRIPTION
**Pull request by Tristancalard**
Add inventory routes Unit tests
Add projects routes Unit tests
Do :
    npm run test

### CODE REVIEW

File: backend/src/test/inventory.test.ts [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+            name: 'name',
+            image: 'image',
+            category: 'category',
+            quantity: 2,
+            available: true,
+            description: 'description',
+        });
+        const res = await chai.request(app).get('/inventory');
+        expect(res).to.have.status(200);
+        expect(res.text).to.equal('[{"id":1,"name":"name","image":"image","category":"category","quantity":2,"available":true,"description":"description"}]');
```
Comment (thmarinho) [inline backend/src/test/inventory.test.ts]:
```suggestion
        const data = {
            id: 1,
            name: 'name',
            image: 'image',
            category: 'category',
            quantity: 2,
            available: true,
            description: 'description',
        };
        await respository.save(data);
        
        const res = await chari.request(app).get('/inventory')
        expect(res).to.have.status(200)
        expect(res.body).to.beDefined()
        expect(res.body.id).to.be(data.id)
        [...]
```

**Review by thmarinho — CHANGES_REQUESTED** [review]

File: backend/src/test/inventory.test.ts [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        repository = dataBase.getRepository(Inventory);
+    });
+
+    afterEach(async () => {
+        await repository.query('DELETE from inventory;');
+    });
+
+    it('should return all items from the inventory on /inventory GET', async () => {
+        const data = {
+            id: 1,
```
Comment (thmarinho) [inline backend/src/test/inventory.test.ts]: delete this

File: backend/src/test/inventory.test.ts [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+            description: 'description',
+        };
+        await repository.save(data);
+
+        const res = await chai.request(app).get('/inventory');
+        expect(res).to.have.status(200);
+        expect(res.body).to.not.be.undefined;
+        expect(res.body).to.be.an('array').that.is.not.empty;
+
+        expect(res.body[0].id).to.equal(Number(data.id));
```
Comment (thmarinho) [inline backend/src/test/inventory.test.ts]: .to.be.defined

File: backend/src/test/inventory.test.ts [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+                console.log('Error connecting to database', err);
+            });
+        repository = dataBase.getRepository(Inventory);
+    });
+
+    afterEach(async () => {
+        await repository.query('DELETE from inventory;');
+    });
+
+    it('should return all items from the inventory on /inventory GET', async () => {
```
Comment (thmarinho) [inline backend/src/test/inventory.test.ts:27]: test with multiple values, this is a get all route

**Review by thmarinho — CHANGES_REQUESTED** [review]
Comments should be applied everywhere

File: backend/src/test/inventory.test.ts (hunk 3 above)
Comment (Tristancalard) [reply backend/src/test/inventory.test.ts]: don't exist in chai

File: backend/src/test/inventory.test.ts [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+            description: 'description',
+        };
+        await repository.save(data2);
+
+        const res = await chai.request(app).get('/inventory');
+        expect(res).to.have.status(200);
+        expect(res.body).to.not.be.undefined;
+        expect(res.body).to.be.an('array').that.is.not.empty;
+
+        expect(res.body[0].id).to.equal(1);
```
Comment (thmarinho) [inline backend/src/test/inventory.test.ts]: to.not.be.undefined

File: backend/src/test/inventory.test.ts [hunk 6]
```diff
@@ -0,0 +1,197 @@
...
+            name: 'name',
+            image: 'image',
+            category: 'category',
+            quantity: 2,
+            available: true,
+            description: 'description',
+        };
+        await repository.save(data);
+
+        const res = await chai.request(app).get('/inventory/3');
```
Comment (thmarinho) [inline backend/src/test/inventory.test.ts]: Do not expect it to be id=3, look at the repository.save return value

File: backend/src/test/inventory.test.ts
Comment (thmarinho) [inline backend/src/test/inventory.test.ts:1]: You should not use hardcoded ids.
If a test is added in the middle of this file, everything will blow up

**Review by thmarinho — CHANGES_REQUESTED** [review]
Comments should applied on each file

**Review by thmarinho — CHANGES_REQUESTED** [review]
`id++` è.é

**Review by thmarinho — APPROVED** [review]

**Pull request merged**